backoffice/src/Repositories/ImageRepository.cs

```diff
@@ -0,0 +1,91 @@
+using Droits.Clients;
+using Droits.Data;
+using Droits.Exceptions;
+using Droits.Helpers;
+using Droits.Models.Entities;
+using Droits.Models.FormModels;
+using Droits.Services;
+using Microsoft.EntityFrameworkCore;
+
+namespace Droits.Repositories;
+
+public interface IImageRepository
+{
+    Task<Image> AddAsync(Image image);
+    Task<Image> UpdateAsync(Image image);
+    Task<Image> GetImageAsync(Guid id);
+    Image GetRandomImage();
+    Task UploadImageFileAsync(Guid id, IFormFile imageFile);
+    Task<Stream> GetImageStreamAsync(string key);
+    Task DeleteImagesForWreckMaterialAsync(Guid wmId, IEnumerable<Guid> imagesToKeep);
+}
+
+public class ImageRepository : BaseEntityRepository<Image>, IImageRepository
+{
+    private readonly IS3Client _client; 
+    public ImageRepository(DroitsContext dbContext, ICurrentUserService currentUserService, IS3Client client) : base(dbContext,currentUserService)
+    {
+        _client = client;
+    }
+    
+    public async Task<Image> GetImageAsync(Guid id)
+    {
+        var image = await Context.Images
+            .FirstOrDefaultAsync(i => i.Id == id);
+
+        if ( image == null )
+        {
+            throw new ImageNotFoundException();
+        }
+
+        return image;
+    }
+    
+    public Image GetRandomImage()
+    {
+        var image = Context.Images
+            .First();
+
+        if ( image == null )
+        {
+            throw new ImageNotFoundException();
+        }
+
+        return image;
+    }
+
+
+    public async Task UploadImageFileAsync(Guid id, IFormFile imageFile)
+    {
+        var key = $"{id}_{Guid.NewGuid()}.{imageFile.FileName}";
+      
+        try
+        {
+            await using var stream = imageFile.OpenReadStream();
+            await _client.UploadImageAsync(key,stream);
+            var image = await GetImageAsync(id);
+            image.Filename = imageFile.FileName;
+            image.FileContentType = imageFile.ContentType;
+            image.Key =  key;
+            await UpdateAsync(image);
+        }
+        catch ( Exception e )
+        {
+            Console.WriteLine(e);
+            throw;
+        }
+    }
+
+
+    public async Task<Stream> GetImageStreamAsync(string key) => await _client.GetImageAsync(key);
+    
+    public async Task DeleteImagesForWreckMaterialAsync(Guid wmId, IEnumerable<Guid> imagesToKeep)
+    {
+        var images = await Context.Images
+            .Where(image => image.WreckMaterialId == wmId && !imagesToKeep.Contains(image.Id))
+            .ToListAsync();
+
+        Context.Images.RemoveRange(images);
```

Comment:
Alright one more thing. 

Once we have imagesToDelete (as a list if images)

*Before* we remove the range from the db, we want to pass it to a _s3Client function to remove the files (maybe)

loop through the list and delete all the keys from the bucket. 

We can ask RoW about data retention etc on this (or rather you can)

But if we have the code, we can then change it, like maybe it's more of a soft delete (move the files to a deleted-images folder within images or something

which raises another question...